Source/ProgressionSystemRuntime/Private/LevelActors/PSStarActor.cpp

```diff
@@ -45,44 +59,96 @@ void APSStarActor::OnLocalCharacterReady(APlayerCharacter* Character, int32 Char
+{
+	InitStarMenuAnimation();
+}
+
 // Is called when any cinematic started
 void APSStarActor::OnAnyCinematicStarted(const UObject* LevelSequence, const UObject* FromInstigator)
+{
+	SetAnimationStartTime(StartTimeHideStarsInternal);
+	TryPlayHideStarAnimation();
+}
+
+// Is called to set a start time for animations
+void APSStarActor::SetAnimationStartTime(float& StartTime)
 {
 	const UWorld* World = GetWorld();
 	check(World);
 
-	StartTimeHideStarsInternal = World->GetTimeSeconds();
+	StartTime = World->GetTimeSeconds();
 
-	// Enable tick and process this frame
+	// Enable tick for the actor
 	SetActorTickEnabled(true);
-	TryPlayHideStarAnimation();
 }
 
-// Hiding stars with animation
-void APSStarActor::TryPlayHideStarAnimation()
+// Called to initialize the Star menu animation
+void APSStarActor::InitStarMenuAnimation()
 {
-	if (!StartTimeHideStarsInternal)
+	if (ensureMsgf(UMyBlueprintFunctionLibrary::GetMyGameState()->GetCurrentGameState(), TEXT("%s: 'GetMyGameState' is null"), *FString(__FUNCTION__))
+		&& UMyBlueprintFunctionLibrary::GetMyGameState()->GetCurrentGameState() == ECurrentGameState::Menu)
 	{
-		return;
+		SetAnimationStartTime(StartTimeMenuStarsInternal);
+		TryPlayMenuStarAnimation();
+	}
+	else
+	{
+		SetActorTickEnabled(false);
+		StartTimeMenuStarsInternal = 0.f;
 	}
+}
+
+// Hiding stars with animation in main menu when cinematic is start to play
+void APSStarActor::TryPlayHideStarAnimation()
+{
+	const FPSRowData& CurrentRow = UPSWorldSubsystem::Get().GetCurrentRow();
+	TryPlayStarAnimation(StartTimeHideStarsInternal, CurrentRow.HideStarsAnimation);
+}
 
+// Menu stars with animation in main menu idle 
+void APSStarActor::TryPlayMenuStarAnimation()
+{
 	const FPSRowData& CurrentRow = UPSWorldSubsystem::Get().GetCurrentRow();
-	if (!CurrentRow.HideStarsAnimation)
+	TryPlayStarAnimation(StartTimeMenuStarsInternal, CurrentRow.MenuStarsAnimation);
+}
+
+// Playing star animation
+void APSStarActor::TryPlayStarAnimation(float& StartTime, UCurveTable* AnimationCurveTable)
+{
+	if (!StartTime || !AnimationCurveTable)
 	{
 		SetActorTickEnabled(false);
-		StartTimeHideStarsInternal = 0.f;
+		StartTime = 0.f;
 		return;
 	}
 
-	const float SecondsSinceStart = GetWorld()->GetTimeSeconds() - StartTimeHideStarsInternal;
+	const float SecondsSinceStart = GetWorld()->GetTimeSeconds() - StartTime;
 
-	const bool bIsFinished = !UGameplayUtilsLibrary::ApplyTransformFromCurveTable(this, CurrentRow.HideStarsAnimation, SecondsSinceStart);
+	const bool bIsFinished = !UGameplayUtilsLibrary::ApplyTransformFromCurveTable(this, AnimationCurveTable, SecondsSinceStart);
 	if (bIsFinished)
 	{
-		SetActorTickEnabled(false);
-		StartTimeHideStarsInternal = 0.f;
+		// if it's looped menu star animation then reset
+		// if it's one-time cinematic animation - disable tick and stop
+		if (AnimationCurveTable == UPSWorldSubsystem::Get().GetCurrentRow().MenuStarsAnimation)
+		{
+			StartTime = 0.f;
```

Comment:
your animation in menu does not start from the beginning of this like `StartTime = 0.f` meaning you disable animation. Instead, for Menu Animation, you should do  `StartTime = GetWorld()->GetTimeSeconds();`, so it will start from the beginning always looping. No need for improvements in `ApplyTransformFromCurveTable` as it works as expected

Reply:
done as part of: https://github.com/h4rdmol/ProgressionSystem/pull/14/commits/6141759e381d2b8658971efac878aaebce1a9cf7